core-tests/src/test/java/org/mule/runtime/core/internal/processor/strategy/AbstractProcessingStrategyTestCase.java

```diff
@@ -720,6 +719,41 @@ public RejectingScheduler(Scheduler delegate) {
     }
   }
 
+  static class RetryScheduler extends TestScheduler {
+
+    private final Scheduler delegate;
+
+    public RetryScheduler(Scheduler delegate) {
+      super(1, "prefix", true);
+      this.delegate = delegate;
+    }
+
+    private Future<?> submit(Supplier<Future<?>> futureSupplier) {
+      while (!delegate.isShutdown() && !delegate.isTerminated()) {
+        try {
+          return futureSupplier.get();
+        } catch (RejectedExecutionException ree) {
+          try {
+            Thread.sleep(5);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7624&issues=AWh8XXdPotIXnmsUOEzC&open=AWh8XXdPotIXnmsUOEzC)

[ISSUE_KEY]: AWh8XXdPotIXnmsUOEzC